### DISCUSSION TITLE
Perform RDREC Operation Using Library

### DISCUSSION THREAD

**Discussion by mason3k** [gijzelaerr/python-snap7#325] [Q&A] [answered]
Hello,

Later models of S7 devices (1200 and 1500) begin using "Instructions" in place of SZL requests. Is there any way to perform these instruction sets via the Snap-7 library? Specifically, I'm interested in the possibility of performing a RDREC Instruction (SFB 52) using python-snap7. If this is not possible, I suppose you would need read the values into global data blocks on the PLC and then pull from there.

Thanks!

**Comment by gijzelaerr** [comment, marked as answer]
RDREC (SFB 52) is a PLC-internal instruction that runs on the PLC CPU itself -- it cannot be invoked remotely via the S7 communication protocol. The correct approach is to have PLC logic execute RDREC and store the results in a global data block, then use client.db_read() to retrieve that data remotely.
